core/break.lua

```diff
@@ -74,34 +75,73 @@ function lineBreak:trimGlue() -- 842
   nodes[#nodes+1] = SILE.nodefactory.penalty(inf_bad)
 end
 
+-- NOTE FOR DEVELOPERS: this method is called when the linebreak.parShape
+-- setting is true. The arguments passed are self (the current line) and a
```

Comment:
`self` is the linebreaker instance (rather than the current line)

Reply:
Roger that. I'm also fixing LTR vs RTL handling of extra space when some shape values are left unspecified.

Reply:
by unspecified, ypu mean when some is "nil" as stated here:
> The expected return is three values, any of which may be nil to use default values or a measurement to override the defaults. The values are considered as left, width, and right respectively.

I am unsure what default values are and where it is handled (should it even have defaults?)

Reply:
Defaults are `0, self.hsize, 0`, but you don't have to return those and they aren't configurable except by supplying a `parShape()` function that does something different. Anything left `nil` will default to those with the addition that changing any of them also makes the appropriate adjustment to the defaults. For example setting a non-zero left with reduces the width from `self.hsize` to `self.hsize - left`.